### ISSUE TITLE
I think suppressors need to be reworked.

### ISSUE THREAD

**Issue by Nhatorama** [CleverRaven/Cataclysm-DDA#22805]
**Game version:**
0.C-25994-g7edf492817
**Operating system:**
Manjaro Linux 64-bit
**Tiles or curses:**
Tiles
**Mods active:**
  "dda",
  "Experimental-Z-Level",
  "Medieval_Stuff",
  "More_Survival_Tools",
  "No_Fungi",
  "StatsThroughSkills",
  "blazemod",
  "Tanks",
  "bio_mod",
  "boats",
  "craftgp",
  "deoxymod",
  "filthy_morale",
  "makeshift",
  "more_buildings",
  "more_classes_scenarios",
  "more_locations",
  "national_guard_camp",
  "no_flaming_weapons",
  "no_npc_food",
  "no_rivtech_guns",
  "novitamins",
  "realguns"

### Expected behavior
No damage penalty from using suppressors.
Increase aim speed of suppressed firearms. (optional)
### Actual behavior
Damage penalty for using suppressors.

### I think suppressors need to be reworked.

Does a Silencer Affect Bullet Velocity? 9mm, .22, and .223 suppressed & unsuppressed
https://www.youtube.com/watch?v=-WPHcqr6dcI

Where I think the idea of making a suppressor reduce damage came from.
https://en.wikipedia.org/wiki/Suppressor#Wipes_and_packing_material

_While many suppressors used wipes in the Vietnam War era, most modern suppressors do not use them as anything that touches the projectile has significant accuracy implications. All "wipes" deteriorate quickly and require disassembly and spare parts replacement._

_Wipes, packing materials and purpose-designed wet cans have been generally abandoned in 21st-century suppressor design because they decrease overall accuracy and require excessive cleaning and maintenance._

http://www.thetruthaboutguns.com/2012/05/foghorn/ask-foghorn-does-a-silencer-effect-the-velocity-of-the-bullet/

_While I was at AAC’s headquarters last year I learned about this idea that a silencer would slow down the bullet. Turns out that its based in fact — the early silencers used a rubber gasket (or “wipe”) at the end of the can that would contact the round and slow it down. The idea was that the gasket would contain the gases for a little longer as the bullet flew downrange quieting the gun even more, but the friction wore out the gaskets quickly and proved to decrease the velocity and accuracy of the rounds.
Modern silencers don’t use the wipes anymore. As AAC’s guys tell it, the silencers are machined to such precision that there isn’t much distance between the round and the baffles to begin with, and a wipe is unnecessary._

https://en.wikipedia.org/wiki/Suppressor#Characteristics

_In weapons firing supersonic ammunition, the bullet itself produces a loud and very sharp sound as it leaves the muzzle in excess of the speed of sound and gradually reducing speed as it travels downrange. This is a small sonic boom, and is referred to in the firearm field as "ballistic crack" or "sonic signature". Subsonic ammunition eliminates this sound, but at the cost of lower velocity, resulting in decreased range and much decreased muzzle energy, thus lessening effectiveness on the target._

_9×19mm Parabellum, a very popular caliber for suppressed shooting, can use almost any factory-loaded 147 grains (9.5 g) weight round to achieve subsonic performance. These 147 gr weight bullets typically have a velocity of 900–980 feet per second (270–300 m/s), which is less than the common 1,140 feet per second (350 m/s) speed of sound._

_Instead of using subsonic ammunition, one can lower the muzzle velocity of a supersonic bullet before it leaves the barrel. Some suppressor designs, referred to as "integrals", do this by allowing gas to bleed off along the length of the barrel before the projectile exits. The MP5SD is the best example of this with holes right after the chamber of the barrel used to reduce a regular 115 or 124 gr ammunition to subsonic velocities._

Makeshift Suppressor
https://www.youtube.com/watch?v=7t_pcWPdSDs

In conclusion:

- The added weight and length added to a firearm by an attached suppressor can be a detriment to aim speed.
Increase aiming cost/speed of suppressed firearms.
- Modern suppressors do not slow down bullets as they fly through them, in fact on average a bullet’s speed is increased.
Remove damage penalty from suppressors.

In the game files where can I find the version I’m running that I can easily copy/paste from?

**Comment by aeoo** [comment]
I don't see how to disagree with any of this when it comes to factory-made suppressors.

However, for the DIY suppressors, maybe there should be a choice between either using a wipe or not using it with the result being two different types of DIY suppressors:

Type 1 with the wipe: lower accuracy and damage, but better sound suppression.

Type 2 without the wipe: the same accuracy and damage as a factory-made suppressor, but thanks to lower tolerances from hand-machining the thing, the sound suppression quality is nowhere near as good as a factory suppressor.

For the type 1 DIY suppressor the wipes can be made from rubber hose that people can find in the game.  One rubber hose should be convertible to many wipes.  Maybe 1 to 10.   Wipes should degrade with usage and require replacement.  If one fails to replace a wipe, then a type 1 suppressor should function even slightly worse than a type 2 one I think.  Or maybe these wipes are not replaceable and make the entire suppressor has to be discarded.  Maybe type 1 DIY suppressors can have charges and after those charges run out, they're considered broken and unusable.

Or, a simpler option is to say forget the type 1 stuff entirely, and assume all DIY suppressors are type 2, so that no suppressor negatively affects accuracy and damage, but they differ in their sound suppression qualities and bulkiness.

And then a factory suppressor would *greatly* suppress the sound with practically no downsides short of a small one for added weight and aim speed.  However, increased weight should lower recoil too (hopefully this is already an automatic calculation).

tl:dr; I strongly agree re:factory made ones, but what about the DIY ones?

**Comment by aeoo** [comment]
Also, DIY sound suppressors don't have to be these giant scope-blocking things:

https://www.youtube.com/watch?v=kP_BIb_OHZI

Here's an example of a DIY suppressors that's relatively thin.

**Comment by Nhatorama** [comment]
Honestly making a makeshift suppressor that involves pretty much making a suppressor from scratch doesn't really sound like it follows the definition of _makeshift_.

To me the oil filter suppressor is more makeshift than making a much more elaborate _survivor_ suppressor with wipes/baffles/etc. The only reason why I would make a _survivor_ suppressor is if I'm  playing in a 0 cities world and oil filters are next to impossible to find because there are no cars around that have ever needed oil filters. In a 0 cities world you won't find any fridges to scavenge rubber hoses from, so following:
> [Wipes are inner dividers intended to touch the bullet as it passes through the suppressor, and are typically made of rubber, plastic, or foam.](https://en.wikipedia.org/wiki/Suppressor#Wipes_and_packing_material)

Wipes should be craftable with either rubber hoses or plastic chunks (will require the plastic mold "tool")

So the lineup would be (from most to least effective):
- Suppressor
- Makeshift (oil filter) suppressor OR Oil filter suppressor
- Survivor suppressor

> If one fails to replace a wipe, then a type 1 suppressor should function even slightly worse than a type 2 one I think.

I imagine that the wipes in a neglected survivor suppressor would eventually droop/melt towards the trajectory that the barrel/bore is pointing at and when you least expect it the next bullet you fire hits the drooped/melted wipe, which in turn causes the bullet to create an extra hole in your survivor suppressor in a less flashy way than [this](https://www.youtube.com/watch?v=BczhT1ByrXA).

>Or maybe these wipes are not replaceable and make the entire suppressor has to be discarded.

IMO if you made it, then you can fix/maintain it.

>Or, a simpler option is to say forget the type 1 stuff entirely, and assume all DIY suppressors are type 2, so that no suppressor negatively affects accuracy and damage, but they differ in their sound suppression qualities and bulkiness.

This would get my thumbs up. I would make the survivor suppressor attract half as much attention than a non-suppressed firearm.
A few questions though:
- Is it still hearing safe? Not sure how permanent deafness works in-game, but if not would the survivor suppressor double the amount of rounds you need to shoot (compared to a non-suppressed firearm) before hearing damage occurs?
- How would going full auto with it affect it? Would it make you use double the charges for every full auto round you shot or not have a multiplier so long as you shoot in brief bursts and with breaks in between?

After a quick search for more in depth literature on Vietnam era suppressors, I couldn't find anything. I think [these](http://www.thefirearmblog.com/blog/wp-content/uploads/2017/03/IMG_6657.png) are wipes.

>And then a factory suppressor would greatly suppress the sound with practically no downsides short of a small one for added weight and aim speed.

Not sure if I'm nitpicking, but since real suppressors have next to no disadvantages (please don't exagerate the aim speed modifier) adding an _optional_
cleaning/maintenance task for every # of rounds shot through [pistol caliber suppressors](http://46.101.143.62:8102/#blob:GUNMOD/suppressor_compact) might be called for. I've never come across one, so another unicorn for me. Rifle suppressors are pretty much self-cleaning, so no maintenance required.

>[A good/general number of rounds fired between rimfire can maintenance is 300.](https://blog.silencershop.com/clean-silencer/)

A few messages to the player that they should clean their suppressor:
`The suppressor is running a bit loud. You should probably clean it.`
After 300 or however many rounds is chosen to be the treshold for really needing to clean the suppressor, the current  `Your ears ring.` and its associated hearing damage could be followed.

>However, increased weight should lower recoil too (hopefully this is already an automatic calculation).

I was going to add:

>[discussing low-velocity pistol calibers, suggest the recoil reduction is around 15%.[36] With high-velocity calibers, recoil reduction runs in the range of 20–30%.](https://en.wikipedia.org/wiki/Suppressor#Characteristics)

but then I double checked suppressor's stats and there is indeed a +2 for _suppressor's_ and +1 for _compact suppressor's_ handling modifier. Is the +1 or +2 fairly close to the 15% to 30% recoil reduction? I don't know.

> Also, DIY sound suppressors don't have to be these giant scope-blocking things:

Scopes have been my tastiest unicorns for CDDA. Unless I found a rifle (M107 Barrett) with it installed, I've yet to come across a scope collecting dust. So I've either been extremely unlucky or scopes are really that rare and a player will want to make half a tootpick or a toothpick high iron sights for their oil filter suppressors, unless of course they don't plan on doing anything except point shooting.

**Comment by BorkBorkGoesTheCode** [comment]
mugling tried to fix these: https://github.com/CleverRaven/Cataclysm-DDA/pull/17724

**Comment by BorkBorkGoesTheCode** [comment]
Factory-made suppressors no longer have a damage penalty.

**Comment by thethunderhawk** [comment]
There ought to be room for legitimately Jerry-rigged suppressors just made from rags and piping. I’ve personally watched people fire .22s with little more than a 2-liter bottle full of styrofoam packing peanuts duct taped to the barrel. They *do* have an appreciable affect on both sound and performance.
